index.js

```diff
@@ -932,6 +941,7 @@ window.mobileCheck = function() {
             {
                 AddRemoveUrlParam(false, 'states', stateInt);
                 g_statesSelected.delete(stateInt);
+                GetDistricts();
```

Comment:
Rather than using onAdd(), or onRemove() you can use onShow(), onHide(). This will result in lesser number of API calls.
onAdd()/remove gets called every time a state/district is added. onShow()/hide get called only when user has selected/removed all states he/she wants to and the dropdown closes.
This will not affect functionality only perf.
Just a suggestion. Final call you can take.
We can also make the perf improvement in future and check in the code as is right now.